### PR TITLE
【Prim】Support Relu Custom VJP

### PR DESCRIPTION
**Pull request by JiabinYang**
### PR types

Others
### PR changes

Others
### Describe

Pcard-66975
This PR support relu custom vjp.

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/prim/composite_ops/test_composite_relu_custom_vjp.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    paddle.disable_static()
+    inputs.stop_gradient = False
+    res = fn(inputs)
+
+    gradients = paddle.grad(res, inputs)
+    return gradients
+
+
+class TestCompositeSoftmaxPrimBackward(unittest.TestCase):
+    "test composite softmax and prim backward"
```
Comment (xiaoguoguo626807) [inline python/paddle/fluid/tests/unittests/prim/composite_ops/test_composite_relu_custom_vjp.py:69]: change softmax as relu

File: python/paddle/fluid/tests/unittests/prim/composite_ops/test_composite_relu_custom_vjp.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
+# Copyright (c) 2022 PaddlePaddle Authors. All Rights Reserved.
```
Comment (xiaoguoguo626807) [inline python/paddle/fluid/tests/unittests/prim/composite_ops/test_composite_relu_custom_vjp.py:1]: it's better  to delete this if op test with prim can cover this test

File: python/paddle/incubate/autograd/composite_rules.py [hunk 3]
```diff
@@ -97,6 +97,7 @@ def composite_batchnorm(
     batch_mean = zeros(run_mean.shape, run_mean.dtype)
     batch_var = zeros(run_var.shape, run_var.dtype)
     if not use_run_stat:
+
```
Comment (xiaoguoguo626807) [inline python/paddle/incubate/autograd/composite_rules.py:100]: unnecessary change

**Review by xiaoguoguo626807 — APPROVED** [review]

**Review by cyber-pioneer — APPROVED** [review]
LGTM

File: python/paddle/fluid/tests/unittests/prim/composite_ops/test_composite_relu_custom_vjp.py (hunk 2 above)
Comment (JiabinYang) [reply python/paddle/fluid/tests/unittests/prim/composite_ops/test_composite_relu_custom_vjp.py:1]: Since activation op test only test simple case, add this case can be more safe to use.

File: python/paddle/incubate/autograd/composite_rules.py (hunk 3 above)
Comment (JiabinYang) [reply python/paddle/incubate/autograd/composite_rules.py:100]: pre-commit hook did this

File: python/paddle/fluid/tests/unittests/prim/composite_ops/test_composite_relu_custom_vjp.py (hunk 1 above)
Comment (JiabinYang) [reply python/paddle/fluid/tests/unittests/prim/composite_ops/test_composite_relu_custom_vjp.py:69]: Done in [PR 51838](https://github.com/PaddlePaddle/Paddle/pull/51838)
